include/boost/url/impl/segments_encoded.hpp

```diff
@@ -405,7 +288,7 @@ insert(
             first, last),
         detail::make_enc_segs_iter(
             first, last));
-    return { *u_, before.i_ };
+    return std::next(begin(), before.i_);
```

Comment:
umm.... you are incrementing `before.i_` times?

Reply:
Yep. Because this is not constant, the pointers get invalidated when we change the URL. I think I sent a comment about that earlier today. I can only see two solutions here:

- Recreate and increment `before.i_` times. It's at least not a bad placeholder because the complexity is still linear.
- Reimplement all iterators to make them use offsets instead of pointers.

Reply:
We can always improve it later I guess, without changing the API